### PR TITLE
Use new exception in PluginsArchiver & add previous exceptions to backtrace in fatal error report

### PR DESCRIPTION
**Pull request by diosmosis**
Refs #13466 

Changes:
* When wrapping thrown exception, use custom exception class since thrown exception may not have compatible constructor.
* Add previous exceptions to fatal error info's backtrace property.
* Recongize previous exceptions in API responses as well (so they will appear in core:archive output).

### CODE REVIEW

File: core/ArchiveProcessor/PluginsArchiver.php [hunk 1]
```diff
@@ -177,15 +177,7 @@ public function callAggregateAllPlugins($visits, $visitsConverted, $forceArchivi
...
-                    $className = get_class($e);
-
-                    if ($className === 'PHPUnit_Framework_Exception' || (class_exists('PHPUnit_Framework_Exception', false) &&  is_subclass_of($className, 'PHPUnit_Framework_Exception'))) {
-                        $exception = new $className($e->getMessage() . " - caused by plugin $pluginName", $e->getCode(), $e->getFile(), $e->getLine(), $e);
-                    } else {
-                        $exception = new $className($e->getMessage() . " - caused by plugin $pluginName", $e->getCode(), $e);
-                    }
-
-                    throw $exception;
+                    throw new PluginsArchiverException($e->getMessage() . " - in plugin $pluginName", $e->getCode(), $e);
```
Comment (tsteur) [inline core/ArchiveProcessor/PluginsArchiver.php]: I suppose that should work. Although wondering if anything may trigger like an `NoAccessException` which should maybe show login page etc. I don't think it's an actual issue though but maybe could be. Not sure if we handle some other exceptions in any specific way.

Comment (diosmosis) [reply core/ArchiveProcessor/PluginsArchiver.php]: we could use reflection to change the `$message` property of the exception. Not sure how future proof that is, but I suppose the tests will tell us quickly if it breaks.

Comment (tsteur) [reply core/ArchiveProcessor/PluginsArchiver.php]: Could work as well I suppose. I reckon it'd be just like https://stackoverflow.com/a/29542873 

Although I reckon the current approach is fine as well, just not sure if any specific exceptions would be treated differently.

Comment (diosmosis) [reply core/ArchiveProcessor/PluginsArchiver.php]: @tsteur Does that mean this PR is ok to merge or do you want to think about it some more?

Comment (tsteur) [reply core/ArchiveProcessor/PluginsArchiver.php]: I reckon it should be fine 👍

**Pull request merged**